### PR TITLE
(#1642) Update combo box text clear/unselected event

### LINKED ISSUE #1642: Combo Box - Differentiate between text clearing and unselecting a selected value
We need to capture analytics on when users unselect their selection (e.g. clicking the X or pressing esc after selection). Right now there is only a single event for both text being cleared from the input element, and users unselecting their selection. So the proposed change is:
1. change `usa-combo-box:cleared` to `usa-combo-box:input:text-cleared`. This should fire for:
   * only when the text input has had its input cleared out without a selection being made. 
   * DETAILS:
      * any current selection
      * the text that got cleared
3. add `usa-combo-box:unselected`. This should fire when:
    *  a user has clicked the X, or pressed escape, to remove their selection.
    * DETAILS:
       * The previousselection

### PR DESCRIPTION
**Pull request by olitharp-nci**
- Updates some lacking documentation as well

Closes #1642

### CODE REVIEW

File: packages/ncids-js/src/components/usa-combo-box/__tests__/combo-box.events.test.ts [hunk 1]
```diff
@@ -484,5 +496,13 @@ describe('Combo box - Events', () => {
...
 			inputValue: 'aa',
 		});
+
+		await user.keyboard('[Escape]');
+		expect(textClearedEvent).toHaveBeenCalledTimes(1);
+		expect(textClearedEvent.mock.calls[0][0].detail).toEqual({
+			comboBox: comboBoxEl,
+			previousInputValue: 'aa',
+			selected: selected,
+		});
```
Comment (bryanpizzillo) [inline packages/ncids-js/src/components/usa-combo-box/__tests__/combo-box.events.test.ts:506]: Do you need a test for when there is no selection and you clear the text?

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
One thing

File: packages/ncids-js/src/components/usa-combo-box/__tests__/combo-box.events.test.ts (hunk 1 above)
Comment (olitharp-nci) [reply packages/ncids-js/src/components/usa-combo-box/__tests__/combo-box.events.test.ts:506]: After discussion, we're going to deal with this in another ticket.

**Review by bryanpizzillo — APPROVED** [review]
Joe and Dev verified that the analytics can work correctly for the types page. So we will merge this once it is done building.

**Pull request merged**